compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -190,6 +190,23 @@ INTERCEPTOR(int, fcntl, int filedes, int cmd, ...) {
   return REAL(fcntl)(filedes, cmd, arg);
 }
 
+INTERCEPTOR(int, ioctl, int filedes, unsigned long request, ...) {
+  __rtsan_notify_intercepted_call("ioctl");
+
+  // See fcntl for discussion on why we use intptr_t
+  // And why we read from va_args on all request types
+  using arg_type = intptr_t;
+  static_assert(sizeof(arg_type) >= sizeof(struct ifreq *));
+  static_assert(sizeof(arg_type) >= sizeof(int));
+
+  va_list args;
+  va_start(args, request);
+  arg_type arg = va_arg(args, arg_type);
+  va_end(args);
+
+  return REAL(ioctl)(filedes, request, arg);
```

Comment:
is there va_list overload?

Reply:
we do these ugly macro, but we know the size of va_list
https://github.com/llvm/llvm-project/blob/8ffe63fb556915c041e8e9bc2d1bf4325f12ba26/compiler-rt/lib/asan/asan_interceptors.cpp#L375-L381